### PR TITLE
Add new Tips page

### PR DESCRIPTION
**Pull request by chrisgarrity**
Initial version of tips page, uses cards from new resources server (s3)

* added get-localized-urls script to get the generated json from the s3 bucket
* added /tips route
* first version of tips page/styling

Still to do:
* correct localized-urls for other pages
* remove thingstotry
* redirect /go

Notes & Questions:
* l10n-static is still needed for build-locales to know which URLs to include in the bundle, however the actual URL should come from the get-localized-urls generated file. Perhaps l10n-static should just be an array of ids?
* Checking in localized-urls.json even though it's now generated because it may be helpful to debug any problems with translated resources.

### CODE REVIEW

File: src/views/tips/tips.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    getInitialState: function () {
+        return {
+            currentTile: Tiles[0],
+            TTTModalOpen: false
+        };
+    },
+    showTTTModal: function (tile) {
+        // expects translated tile
+        this.setState({currentTile: tile});
+        this.setState({TTTModalOpen: true});
```
Comment (paulkaplan) [inline src/views/tips/tips.jsx]: This is minor, but unless you have a specific reason for needing to do this in two `setState` calls, I think it is better to combine them into a single `this.setState({currentTile: tile, TTTModalOpen: true});`

File: src/views/tips/tips.jsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            };
+            translatedTiles.push(
+                <TTTTile
+                    key={key}
+                    onGuideClick={this.showTTTModal.bind(this, translatedTile)}
+                    {...translatedTile}
+                />
+            );
+        }, this); // don't forget to pass 'this' into map function
+        return translatedTiles;
```
Comment (paulkaplan) [inline src/views/tips/tips.jsx]: I think you can simplify this whole thing by just returning from the `.map`:

```
+        Tiles.map(function (tile, key) {
 +            translatedTile = {
 +                title: formatMessage({id: tile.title}),
 +                description: formatMessage({id: tile.description}),
 +                tutorialLoc: tile.tutorialLoc,
 +                activityLoc: formatMessage({id: tile.activityLoc}),
 +                guideLoc: formatMessage({id: tile.guideLoc}),
 +                thumbUrl: tile.thumbUrl,
 +                bannerUrl: tile.bannerUrl
 +            };
 +            return (
 +                <TTTTile
 +                    key={key}
 +                    onGuideClick={this.showTTTModal.bind(this, translatedTile)}
 +                    {...translatedTile}
 +                />
 +            );
 +        }, this);
```
and removing the var declarations from the top

File: bin/lib/localized-urls.json [hunk 3]
```diff
@@ -1,151 +1 @@
...
...ch2Cards.pdf"},"sl":{"cards.Scratch2CardsLink":"https:/resources.scratch.mit.edu/www/cards/sl/Scratch2Cards.pdf"},"sv":{"cards.Scratch2CardsLink":"https:/resources.scratch.mit.edu/www/cards/sv/Scratch2Cards.pdf"},"zh-tw":{"cards.catchCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/catchCards.pdf","cards.danceCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/danceCards.pdf","cards.fashionCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/fashionCards.pdf","cards.flyCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/flyCards.pdf","cards.hide-seekCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/hide-seekCards.pdf","cards.musicCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/musicCards.pdf","cards.nameCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/nameCards.pdf","cards.petCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/petCards.pdf","cards.pongCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/pongCards.pdf","cards.raceCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/raceCards.pdf","cards.storyCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/storyCards.pdf"}}
```
Comment (paulkaplan) [inline bin/lib/localized-urls.json]: Is this something that might need to get updated by hand? Should it be pretty printed? (Easy way in node is to just get the object and run `JSON.stringify(obj, null, '    ')` with the last argument being the tab character.

Comment (paulkaplan) [inline bin/lib/localized-urls.json]: Woops just noticed it is generated by a script.

File: src/views/tips/tips.jsx (hunk 1 above)
Comment (chrisgarrity) [inline src/views/tips/tips.jsx]: 👍  It's always good to get fresh eyes on code :)

**Comment by chrisgarrity** [comment]
@rschamp @mewtaylor it would be good to get a review from one of you for this PR since it involves the routes and localization process. I've already incorporated Paul's jsx suggestions.

File: src/views/tips/l10n-static.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "guides.FlyGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/FlyGuide.pdf",
+    "guides.Getting-Started-Guide-Scratch2Link": "https:/resources.scratch.mit.edu/www/guides/en/Getting-Started-Guide-Scratch2.pdf",
+    "guides.HideandSeekGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/HideandSeekGuide.pdf",
+    "guides.MusicGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/MusicGuide.pdf",
+    "guides.NameGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/NameGuide.pdf",
+    "guides.PetGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/PetGuide.pdf",
+    "guides.PongGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/PongGuide.pdf",
+    "guides.RaceGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/RaceGuide.pdf",
+    "guides.StoryGuideLink": "https:/resources.scratch.mit.edu/www/guides/en/StoryGuide.pdf"
+}
```
Comment (rschamp) [inline src/views/tips/l10n-static.json]: Nit: newline

File: bin/get-localized-urls [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+var args = process.argv.slice(2);
+
+if (!args.length) {
+    process.stdout.write('A output filename must be specified.\n');
+    process.exit(1);
+}
+var filename = path.resolve(__dirname, 'lib', args.shift());
+
+https.get('https://resources.scratch.mit.edu/localized-urls.json', (res) => {
```
Comment (chrisgarrity) [inline bin/get-localized-urls]: 👍

File: bin/lib/localized-urls.json [hunk 6]
```diff
@@ -1,151 +1 @@
...
...ch2Cards.pdf"},"sv":{"cards.Scratch2CardsLink":"https:/resources.scratch.mit.edu/www/cards/sv/Scratch2Cards.pdf"},"zh-tw":{"cards.ScratchCardsAllLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/ScratchCardsAll.pdf","cards.catchCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/catchCards.pdf","cards.danceCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/danceCards.pdf","cards.fashionCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/fashionCards.pdf","cards.flyCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/flyCards.pdf","cards.hide-seekCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/hide-seekCards.pdf","cards.musicCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/musicCards.pdf","cards.nameCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/nameCards.pdf","cards.petCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/petCards.pdf","cards.pongCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/pongCards.pdf","cards.raceCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/raceCards.pdf","cards.storyCardsLink":"https:/resources.scratch.mit.edu/www/cards/zh-tw/storyCards.pdf"}}
```
Comment (chrisgarrity) [inline bin/lib/localized-urls.json]: As we discussed I'll remove localized-urls.json from the repo once it's published with npm (see https://github.com/LLK/scratch-resources/issues/6)

File: src/views/cards/l10n-static.json [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
-    "cards.musicLink": "/pdfs/cards/MusicCards.pdf",
-    "cards.hideLink": "/pdfs/cards/Hide-and-Seek-Cards.pdf",
-    "cards.storyLink": "/pdfs/cards/StoryCards.pdf",
-    "cards.dressupLink": "/pdfs/cards/DressupCards.pdf",
-    "cards.pongLink": "/pdfs/cards/PongCards.pdf",
-    "cards.danceLink": "/pdfs/cards/DanceCards.pdf",
-    "cards.catchLink": "/pdfs/cards/CatchCards.pdf",
-    "cards.petLink": "/pdfs/cards/PetCards.pdf"
-}
+    "cards.Scratch2CardsLink": "https:/resources.scratch.mit.edu/www/cards/en/Scratch2Cards.pdf",
```
Comment (chrisgarrity) [inline src/views/cards/l10n-static.json]: Just goes to show that the value part of l10n-static is no longer needed. They all get set from the resources.scratch.mit.edu generated localized-urls.

File: bin/lib/localized-urls.json [hunk 8]
```diff
@@ -1,151 +1,113 @@
...
+    "cards.flyCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/flyCards.pdf",
+    "cards.hide-seekCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/hide-seekCards.pdf",
+    "cards.musicCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/musicCards.pdf",
+    "cards.nameCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/nameCards.pdf",
+    "cards.petCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/petCards.pdf",
+    "cards.pongCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/pongCards.pdf",
+    "cards.raceCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/raceCards.pdf",
+    "cards.storyCardsLink": "https:/resources.scratch.mit.edu/www/cards/zh-tw/storyCards.pdf"
+  }
+}
```
Comment (rschamp) [inline bin/lib/localized-urls.json]: All of these have a missing `/`, is that an issue?

Comment (chrisgarrity) [inline bin/lib/localized-urls.json]: Yup - fixed in https://github.com/LLK/scratch-resources/pull/8

**Comment by jwzimmer** [comment]
## Testing

### Update "Things to Try" Page
- [ ] Change path to `/tips`
- [ ] Redirect `/go` to `/tips`
- [ ] Add section above tips content that links to the [Getting Starting](http://vimeo.com/llk/start-scratch) video
- [ ] Add section below tips content that links to both download and purchase the [Scratch Coding Cards](https://www.amazon.com/Scratch-Coding-Cards-Creative-Activities/dp/1593277741)
- [ ] Add section below tips content that links to the [Offline Editor](https://scratch.mit.edu/scratch2download/)
- [ ] Add section below tips content that links to the [Help With Scripts](https://scratch.mit.edu/discuss/7/) forum topic

### Retire "Help" Page
- [ ] Retire current "help" template
- [ ] Redirect `/help` to `/tips`

### Update Navigation
- See #1262

### Update "Things to Try" Page
- [ ] Style/layout should match https://github.com/LLK/scratch-www/issues/1261 
  - [ ] (12 cols) Large/desktop
  - [ ] (8 cols) Tablet
  - [ ] (6 cols) Large phone
  - [ ] (4 cols) Phone
- [ ] Change path to `/tips`
- [ ] Redirect `/go` to `/tips`
- [ ] Add section above tips content that includes links to https://scratch.mit.edu/projects/editor/?tip_bar=getStarted and https://scratch.mit.edu/scratchr2/static/pdfs/help/Getting-Started-Guide-Scratch2.pdf
- [ ] Add section below tips content that links to both download and purchase the [Scratch Coding Cards](https://www.amazon.com/Scratch-Coding-Cards-Creative-Activities/dp/1593277741)
- [ ] Add a starter projects section below the Cards that links to https://scratch.mit.edu/starter_projects/
- [ ] Add section below tips content that links to the [Offline Editor](https://scratch.mit.edu/download/)
- [ ] Add section below tips content that links to the [Help With Scripts](https://scratch.mit.edu/discuss/7/) forum topic

### Localization
- [ ] Things to try cards should be available in multiple languages (try ca and zh_TW)
- [ ] Download all cards is available in multiple languages (ca and zh_TW)
- [ ] /info/cards gets card files from resources.scratch.mit.edu
- [ ] Original scratch2 cards are available in multiple languages (there are about 15 of these, try italian, arabic etc)

**Comment by jwzimmer** [comment]
## Design diffs

- [ ] Style/layout should match https://github.com/LLK/scratch-www/issues/1261 
  - [ ] (12 cols) Large/desktop
  - [ ] (8 cols) Tablet
  - [ ] (6 cols) Large phone
  - [ ] (4 cols) Phone

### 12 columns

screen shot 2017-06-27 at 12 02 04 pm

**Comment by chrisgarrity** [comment]
@jwzimmer I'm fixing a couple of these. I'll let Carl comment on whether it's important to implement the other styles:

1) Tip page is fitting within the 12/8/6 column layout, I think the nav bar is also fitting within the same number of columns, so it's just the wireframe nav that's wider
2) @carljbowman worth adjusting?
3) dividing wavy line - Carl said it was a nice to have, and not to bother if it was tricky css
4) Final Copy (after wireframes) has the new wording
5) Missed that one - will update with #1374 
6) The images are supposed to extend off the side (but with overflow hidden - is that broken on Windows?)
7) Fixing along with #1374 
8) That just happens to be right at the cut point between two layouts - there's no padding at that point
9) Fixing
10) Alignment is being handled by Flexrow - OK to leave centered? @carljbowman

**Comment by carljbowman** [comment]
@chrisgarrity @jwzimmer - Thanks for the notes and reviewing.

1. The alignment on the page is correct. At some point the Nav should break out of the grid. We could remove the `inner` container and set flex box to allow the Nav to open up on larger screens.
2. This is fine. No need to adjust.
3. I thought this might be fun 'curve ball' for @TheGrits to figure out, but def a nice to have.
4. 👍 
5. Good catch
6. As @chrisgarrity - The images are suppose to extend off the screen. @jwzimmer - Could you clarify.
7. 👍 
8. As @chrisgarrity - This is a larger known issue with out responsive breakpoints. Not a blocker.
9. 👍 
10. That's fine. 
11. Same as point 6
12. @chrisgarrity - Not a blocker, but would be good to figure out what's going on. 
13. 👍 dragable

**Comment by chrisgarrity** [comment]
@jwzimmer item 12 was fixed in https://github.com/LLK/scratch-www/issues/1383

**Comment by chrisgarrity** [comment]
Item 15: iOS 7.1.2 is really old at this point. I think we should check on iOS 9.0 or greater. From https://caniuse.com/#feat=flexbox it looks like flexbox support wasn't available on iOS until 9.0.

**Comment by jwzimmer** [comment]
The issues related to this PR that were in this milestone with `needs-qa` have been fixed 👍

**Pull request merged**